### DISCUSSION TITLE
Emoji broken after Arch upgrade

### DISCUSSION THREAD

**Discussion by optimistic5** [qtile/qtile#4282] [Q&A] [answered]
Issue not related to qtile directly, but I hope to receive help here.

After Arch upgrade I have few issues.

1. Fonts not render properly.
Before it was like this
![image](https://github.com/qtile/qtile/assets/25054050/1f3d6087-ff1a-4d92-aa57-b8be0f61d9a7)

It was solved by changing 
```python
font='TerminessTTF Nerd Font',
```
to
```python
font='Terminess Nerd Font',
```

After change of font name
![image](https://github.com/qtile/qtile/assets/25054050/7410993e-076a-469f-8b2d-23dd282b2319)

3. Emoji are not render properly in Qtile and system
Before it was like this (note emoji for network, calendar, etc)
https://gitlab.com/level101/qtile/-/raw/master/examples/qtile-screenshot.png

Now I see this
![image](https://github.com/qtile/qtile/assets/25054050/ba45782b-58b8-4c6c-99da-ecb30b792038)

It also renders wrong in browser
https://gitlab.com/level101/qtile/-/blob/master/config.py#L525
I see like
![image](https://github.com/qtile/qtile/assets/25054050/a8b9ba0e-a44e-4a4b-991d-0a5a5cb7739e)

Even this https://github.com/m-col/qtile-config/blob/master/config.py#L587
I see like
![image](https://github.com/qtile/qtile/assets/25054050/2543ed44-8051-46b5-b3cf-370722fd81f2)

**Any idea how to render this emoji properly?**

**Comment by thingsiplay** [comment, marked as answer]
This has something to do with the recent Nerd Font 3.0 update, which changes internally some paths for those characters: https://github.com/ryanoasis/nerd-fonts/releases/tag/v3.0.0 As Archlinux is on the front with the updates, its usually where you will see it first (or if you use Nerd Fonts directly from source). The only real fix is to replace every single of those emojis with an alternate version or find the other path of it. Or you use an older version of the font, but that's not a fix, more like ignoring the problem.

* https://www.nerdfonts.com/cheat-sheet to find emojis and characters by searching.
* There is also a tool to fix these things automatically, but I did not use it and cannot say how it works: https://github.com/loichyan/nerdfix

**Reply by optimistic5** [reply]
@thingsiplay thank you very much for your reply.
You are absolutely right, I missed to check the changelog for font.
For me it was [solved](https://gitlab.com/level101/qtile/-/commit/8e7c43ed7aa288f2ceecc31878a1ef07e6b50e92) by changing the name of font and changing the unicodes.
Unfortunately at time I solved I did not find site you provide (thank you for that), but I used `fontforge`.
Thank you for such good answer!
